docs/how-to/uc-image-creation.rst

```diff
@@ -0,0 +1,307 @@
+The configuration is an iterative process that is best done together with the expected workload.
+
+Once satisfied with the configurations, continue below to learn how those configurations can be set statically during the image build.
+
+.. _custom-ubuntu-core-image:
+
+Create a custom real-time Ubuntu Core image
+-------------------------------------------
+
+This section shows how to statically set the desired Kernel command-line parameters for the Ubuntu Core system.
+To do this, we need to create a custom gadget snap, create a model assertion, and then build the OS image.
+
+.. admonition:: Project directory
+
+    Start in an empty directory.
+    We refer to this in different parts of the document as our *project directory*.
+
+Create the gadget snap
+----------------------
+
+The `gadget snap`_ documentation is a recommended read before starting.
+
+This is best done by forking an existing reference gadget, then changing it for our purpose.
+For example, there is the `pc gadget`_ which is suitable for most AMD64 platforms, and the `pi gadget`_ which is meant for Raspberry Pis.
+
+Let's create a custom gadget snap based on the core22 pc gadget.
+
+Inside the project directory, clone the specific branch and enter the repository:
+
+.. code-block:: shell
+
+    git clone https://github.com/snapcore/pc-gadget.git --branch=22 --depth=1
+    cd pc-gadget
+
+
+Add the desired kernel command line in an array to ``kernel-cmdline.append`` in ``gadget/gadget-amd64.yaml``.
+For example:
+
+.. code-block:: yaml
+
+    kernel-cmdline:
+        append:
+            - nohz=on
+            - nohz_full=2-N
+            - irqaffinity=0-1
+
+
+Refer to :doc:`../reference/kernel-boot-parameters` for the list of supported parameters.
+
+Modify ``snapcraft.yaml`` to fit your application.
+At least, make sure to change the name and version to something distinct, for example, to ``realtime-pc`` and ``example`` respectively.
+
+
+Now, build the gadget snap:
+
+.. code-block:: console
+
+    $ snapcraft --verbose
+    ...
+    Created snap package realtime-pc_example_amd64.snap
+
+
+.. tip::
+    You need to rebuild the snap every time you change the `snapcraft.yaml` file.
+
+
+Create the model assertion
+--------------------------
+
+Create the model assertion inside the project directory.
+Follow the same steps in :ref:`model-assertion` section but replace the ``pc`` snap entry with the following:
+
+.. code-block:: json
+
+    {
+        "name": "realtime-pc",
+        "type": "gadget"
+    },
+
+Unlike the original ``pc`` snap definition, this entry has no listed ``channel`` and ``id``, because it isn't in a Store.
+We have built it locally in the earlier steps and will later on pass it directly to the image builder.
+In practice, the gadget snap should be uploaded to a Store and then listed in the model assertion along with its channel and id.
```

Comment:
Could we use something else besides "in practice" since the rest of this sounds like a best practice recommendation? E.g. "For production use, the gadget snap should be uploaded..."